include/lttng/ust-events.h

```diff
@@ -110,7 +110,12 @@ struct lttng_enum_value {
 struct lttng_enum_entry {
 	struct lttng_enum_value start, end; /* start and end are inclusive */
 	const char *string;
-	char padding[LTTNG_UST_ENUM_ENTRY_PADDING];
+	union {
+		struct {
+			unsigned int is_auto:1;
```

Comment:
I'm tempted to use uint8_t here instead. Bitfields may differ between compiler implementations, and I'm tempted not to use them for an ABI between probe providers and UST.